### PR TITLE
create event & some validations

### PR DESCRIPTION
**Pull request by nicolascr181**

### CODE REVIEW

**Review by sergiomonz — APPROVED** [review]

File: services/events/functions/create/main.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import json
+import boto3
+from decimal import Decimal
+
+client = boto3.resource('dynamodb')
+table = client.Table("events")
+
+def create_event(event=None):
+    
+    # this will search for dynamoDB table
```
Comment (1701722767) [inline services/events/functions/create/main.py]: ??

File: services/events/functions/create/main.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            "event_dateend": event['event_dateend']
+        }
+        validate(new_event)
+        
+    except Exception as e:
+        response['error']  = True
+        response['message'] = str(e).replace("'","")  + " missing"
+        return response
+    
+    new_event = json.loads(json.dumps(new_event), parse_float=Decimal)
```
Comment (1701722767) [inline services/events/functions/create/main.py]: why this code not have a try catch?

File: services/events/functions/create/main.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        "event_max": event['event_max'],
+        "event_datestart": event['event_datestart'],
+        "event_dateend": event['event_dateend']
+    })
+   
+    return response
+
+def validate(data):
+    for name in data.keys():
+        if not data[name]:
```
Comment (1701722767) [inline services/events/functions/create/main.py:70]: this always is False

**Review by 1701722767 — CHANGES_REQUESTED** [review]

File: services/events/functions/create/main.py (hunk 3 above)
Comment (nicolascr181) [reply services/events/functions/create/main.py:70]: This function raises an exception if it happens, not a true or false value

Comment (1701722767) [reply services/events/functions/create/main.py:70]: but no validate that exist some keys, only validate that keys of a data exist in data

File: services/events/functions/create/main.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        raise Exception(str(e))
+
+def validate(data):
+    for name in data.keys():
+        if not data[name]:
+            print(name," missing")
+            raise Exception(str(name))
+            
+def lambda_handler(event, context):
+    # TODO implement
```
Comment (1701722767) [inline services/events/functions/create/main.py]: ??

**Review by 1701722767 — CHANGES_REQUESTED** [review]

File: services/events/functions/create/main.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+# UUID, Universal Unique Identifier, is a python library which helps
+# in generating random objects of 128 bits as ids. 
+# It provides the uniqueness as it generates ids on the basis of time, Computer hardware (MAC etc.).
+def create_id():
+    # new_id = "E" + str(uuid.uuid1())
+    
+    # Generate a random UUID.
+    # probability of collision really really small
+    new_id = "E" + str(uuid.uuid4())
```
Comment (1701722767) [inline services/events/functions/create/main.py]: return the value directly, for not lose memory in this new var

File: services/events/functions/create/main.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+            "range_age": event['range_age'],
+            "price": event['price'],
+            "slots": event['slots'],
+            "max": event['max'],
+            "datestart": event['datestart'],
+            "dateend": event['dateend']
+        })
+        validate_dynamodb_response(response)
+        
+    except Exception as e:
```
Comment (1701722767) [inline services/events/functions/create/main.py]: why catcha if thrown again

File: services/events/functions/create/main.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+def validate(data):
+    for name in data.keys():
+        if not data[name]:
+            print(name," missing")
+            raise KeyError(str(name))
+
+#Verify response from dynamodb
+def validate_dynamodb_response(response):
+    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
+        print("Error al intentar crear un nuevo evento")
```
Comment (1701722767) [inline services/events/functions/create/main.py]: in the general try you can print this errors and evit this

File: services/events/functions/create/main.py [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+            "dateend": event_data['dateend']
+        }
+        validate(new_event)
+        create_event(new_event)
+       
+        
+    except KeyError as e:
+        message['error']  = True
+        message['message'] = KEY_ERROR_MESSAGE[e.args[0]]
+        response['statusCode'] = 500
```
Comment (1701722767) [inline services/events/functions/create/main.py]: this can be a 400 status code

File: services/events/functions/create/main.py [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+        
+    except Exception as e:
+        message['error']  = True
```
Comment (1701722767) [inline services/events/functions/create/main.py:124]: add print e

**Review by 1701722767 — CHANGES_REQUESTED** [review]

File: services/events/functions/create/main.py [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+    "category_id": "La categoría no existe o no fue seleccionada",
+    "user_id" : "No se indicó el usuario al cual le pertenece el evento",
+    "name": "Debe escribir el nombre del evento",
+    "coordinates": "Es necesaria la ubicación del evento",
+    "images": "El evento debe tener imagenes",
+    "description": "El evento debe tener una descripción",
+    "range_age": "El rango de edad debe ser selecionado",
+    "price": "Debe indicar el precio del evento",
+    "slots": "Debe indicar los cupos disponibles actuales del evento",
+    "max": "Debe ingresar la cantidad maxima de personas al evento",
```
Comment (1701722767) [inline services/events/functions/create/main.py:18]: máxima

**Review by 1701722767 — APPROVED** [review]

**Pull request merged**